### PR TITLE
Get api categories

### PR DESCRIPTION
**Pull request by Cnmoosavinia**
Had an issue where I didn't pull after project-setup was approved so have some extra commits as I was trying to correct this. First commit was just to setup the files -> final commit is the tests + solutions

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+const testData = require("../db/data/test-data");
+const app = require("../app.js");
+const connection = require("../db/connection.js");
+
+beforeEach(() => {
+  return seed(testData);
+});
+afterAll(() => {
+  connection.end();
+});
```
Comment (pkaodev) [inline __tests__/app.test.js]: you should be returning connection.end()

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+describe("GET /api/categories", () => {
+  test("GET: 200 - responds with a 200 status and an array of objects containing slug and description properties", () => {
+    return request(app)
+      .get("/api/categories")
+      .expect(200)
+      .then(({ body }) => {
+        const { categories } = body;
+        expect(categories).toHaveLength(4);
+        expect(categories).toBeInstanceOf(Array);
+        categories.forEach((category) => {
```
Comment (pkaodev) [inline __tests__/app.test.js]: line 22 is redundant as you are using a forEach on line 23 (which will only work on an Array)

File: app.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+const express = require("express");
+const app = express();
+const { getCategories } = require("./controller/games.controller.js");
+
+app.use(express.json());
+
```
Comment (pkaodev) [inline app.js]: line 5 is not needed for GET requests. remove it.  only add code which is needed for a ticket.

File: __tests__/app.test.js (hunk 1 above)
Comment (Cnmoosavinia) [reply __tests__/app.test.js]: added return

**Comment by pkaodev** [comment]
Looks good overall!  Please make the changes from the comments I've added and then you are free to merge.

File: __tests__/app.test.js (hunk 2 above)
Comment (Cnmoosavinia) [reply __tests__/app.test.js]: removed toBeInstanceOf(Array) line

File: app.js (hunk 3 above)
Comment (Cnmoosavinia) [reply app.js]: removed this as well

**Comment by Cnmoosavinia** [comment]
thanks very much, made those amendments so will commit and merge, cheers

**Pull request merged**